src/pyupgrade/plugins/os_error_alias.rs

```diff
@@ -0,0 +1,232 @@
+pub trait OSErrorAliasChecker {
+    fn check_error(&self, checker: &mut Checker)
+    where
+        Self: Sized;
+}
+
+impl OSErrorAliasChecker for &Vec<Excepthandler> {
+    fn check_error(&self, checker: &mut Checker) {
+        // Each separate except block is a separate error and fix
+        for handler in self.clone().iter() {
+            handle_except_block(checker, handler);
+        }
+    }
+}
+
+impl OSErrorAliasChecker for &Box<Located<ExprKind>> {
+    fn check_error(&self, checker: &mut Checker) {
+        let mut replacements: Vec<String>;
+        let mut before_replace: Vec<String>;
+        match &self.node {
+            ExprKind::Name { id, .. } => {
+                (replacements, before_replace) = check_module(checker, self);
+                if replacements.is_empty() {
+                    let new_name = get_correct_name(&id);
+                    replacements.push(new_name);
+                    before_replace.push(id.to_string());
+                }
+            }
+            ExprKind::Attribute { .. } => {
+                (replacements, before_replace) = check_module(checker, self);
+            }
+            _ => return,
+        }
+        handle_making_changes(checker, self, before_replace, replacements);
+    }
+}
+
+impl OSErrorAliasChecker for &Located<ExprKind> {
+    fn check_error(&self, checker: &mut Checker) {
+        let mut replacements: Vec<String>;
+        let mut before_replace: Vec<String>;
+        let change_target: &Located<ExprKind>;
+        match &self.node {
+            ExprKind::Name { id, .. } => {
+                change_target = self;
+                (replacements, before_replace) = check_module(checker, self);
+                if replacements.is_empty() {
+                    let new_name = get_correct_name(&id);
+                    replacements.push(new_name);
+                    before_replace.push(id.to_string());
+                }
+            }
+            ExprKind::Attribute { .. } => {
+                change_target = self;
+                (replacements, before_replace) = check_module(checker, self);
+            }
+            ExprKind::Call { func, .. } => {
+                change_target = &func;
+                match &func.node {
+                    ExprKind::Name { id, .. } => {
+                        (replacements, before_replace) = check_module(checker, &func);
+                        if replacements.is_empty() {
+                            let new_name = get_correct_name(&id);
+                            replacements.push(new_name);
+                            before_replace.push(id.to_string());
+                        }
+                    }
+                    ExprKind::Attribute { .. } => {
+                        (replacements, before_replace) = check_module(checker, &func);
+                    }
+                    _ => return,
+                }
+            }
+            _ => return,
+        }
```

Comment:
I'm wondering if this whole part (handling try, raise, etc.) could be changed to instead look at `ExprKind::Attribute` and `ExprKind::Name` (that is: in `ast.rs`, call this plugin for every `ExprKind::Attribute` and every `ExprKind::Name`).

Like, ultimately, we just want to rewrite all references. It doesn't matter if they occur in try, or raise, or wherever. And here, you're just delegating to checks on attribute and name anyway. (Within call, you're then checking name and attribute.)

Would that work? Would it simplify things?

Reply:
I think you are right here... Will take me a minute to confirm but im on it.

Reply:
I’m guessing pyupgrade doesn’t do this, but I’d have to look. If not, I wonder why.

Reply:
So there is one issue with this implementation. If the exceptions are in a tuple, we lose our ability to intelligently group them if we only go by name and attribute (unless there is something I am missing here).

Reply:
Ahhh interesting. We actually have a different rule, in flake8-bugbear, that detects and removes duplicate exceptions. So perhaps we can accept that limitation here, and rely on the other rule to clean up those rare errors? What do you think? It feels like we’re accepting a lot of complexity here to accommodate that.

Reply:
I have one idea I like a little more. Let me see if I can get it to work, and if not I will do your way.

Reply:
I can also give the code a closer read now that I understand that limitation.

Reply:
Haha still thinking through this. The issue with the solution you provided is that if the simplifier ends up with only one option the final value will show as (OSError,) when it should be OSError per pyupgrade. This means we will then have to run a third check. So I am weighing three checks, vs how I can simplify the current code.

Reply:
```
            ExprKind::Name{ .. } | ExprKind::Attribute{ .. } => {
                handle();
            }
```
Thinking about doing something like this. That way we can reduce having the same code copied everywhere.

Reply:
It's ok with me if it ends up as `(OSError,)`.

Reply:
If helpful.

Reply:
Additionally, if we figure out how to get rid of the box I can remove one of the three implementations of OSErrorAliasChecker.

Reply:
```diff --git a/src/checkers/ast.rs b/src/checkers/ast.rs
index b0e48f5d..73e39976 100644
--- a/src/checkers/ast.rs
+++ b/src/checkers/ast.rs
@@ -1087,7 +1087,7 @@ where
                 }
                 if self.settings.enabled.contains(&CheckCode::UP024) {
                     if let Some(item) = exc {
-                        pyupgrade::plugins::os_error_alias(self, item);
+                        pyupgrade::plugins::os_error_alias(self, &**item);
                     }
                 }
             }
diff --git a/src/pyupgrade/plugins/os_error_alias.rs b/src/pyupgrade/plugins/os_error_alias.rs
index f6032b6a..32c9a0a8 100644
--- a/src/pyupgrade/plugins/os_error_alias.rs
+++ b/src/pyupgrade/plugins/os_error_alias.rs
@@ -1,5 +1,5 @@
 use itertools::Itertools;
-use rustpython_ast::{Excepthandler, ExcepthandlerKind, ExprKind, Located};
+use rustpython_ast::{Excepthandler, ExcepthandlerKind, Expr, ExprKind, Located};
 
 use crate::ast::helpers::match_module_member;
 use crate::ast::types::Range;
@@ -162,29 +162,7 @@ impl OSErrorAliasChecker for &Vec<Excepthandler> {
     }
 }
 
-impl OSErrorAliasChecker for &Box<Located<ExprKind>> {
-    fn check_error(&self, checker: &mut Checker) {
-        let mut replacements: Vec<String>;
-        let mut before_replace: Vec<String>;
-        match &self.node {
-            ExprKind::Name { id, .. } => {
-                (replacements, before_replace) = check_module(checker, self);
-                if replacements.is_empty() {
-                    let new_name = get_correct_name(&id);
-                    replacements.push(new_name);
-                    before_replace.push(id.to_string());
-                }
-            }
-            ExprKind::Attribute { .. } => {
-                (replacements, before_replace) = check_module(checker, self);
-            }
-            _ => return,
-        }
-        handle_making_changes(checker, self, before_replace, replacements);
-    }
-}
-
-impl OSErrorAliasChecker for &Located<ExprKind> {
+impl OSErrorAliasChecker for &Expr {
     fn check_error(&self, checker: &mut Checker) {
         let mut replacements: Vec<String>;
         let mut before_replace: Vec<String>;
```

Reply:
Push that real quick!! I can add my simplified function to it.

Reply:
Still more complicated, but I can add a bunch of comments so its crystal clear.

Reply:
> Push that real quick!! I can add my simplified function to it.

Nvmd I got it